server_options.go

```diff
@@ -176,3 +177,11 @@ func WithServerJSONSkipDefaults(skipDefaults bool) ServerOption {
 		o.JSONSkipDefaults = skipDefaults
 	}
 }
+
+// WithServerJSONCamelCaseNames configures JSON serialization to use
+// lowerCamelCase field names rather than proto snake_case names.
+func WithServerJSONCamelCaseNames() ServerOption {
+	return func(o *ServerOptions) {
+		o.JSONCamelCaseNames = true
+	}
+}
```

Comment:
The option should take a boolean value, even if the default is false. This is to stay consistent with other options, and to allow overrides/reset if called multiple times.

```
```go
// WithServerJSONCamelCaseNames configures JSON serialization to use 
// lowerCameCase field names rather than the original proto field names.
// It is disabled by default, because JSON is commonly used for manual
// debugging. But sometimes converting to lowerCamelCase is needed
// to match the default canonical encoding on other proto-json parsers.
// See: https://developers.google.com/protocol-buffers/docs/proto3#json
func WithServerJSONCamelCaseNames(val bool) ServerOption {
	return func(o *ServerOptions) {
		o.JSONCamelCaseNames = val
	}
}
```